scanner/api/database.cpp

```diff
@@ -297,6 +307,7 @@ Result Database::new_table(const std::string& table_name,
     storage_.get(), internal::DatabaseMetadata::descriptor_path());
 
   i32 table_id = meta.add_table(table_name);
+  assert(table_id != -1);
```

Comment:
This is fine for now, but moving forward we should replace asserts with `LOG_IF(FATAL)` so we can leave reasoning why the assert shouldn't fail.